lib/common/strings.c

```diff
@@ -896,12 +893,13 @@ pcmk__str_in_list(GList *lst, const gchar *s, uint32_t flags)
 {
     GCompareFunc fn;
 
-    if (lst == NULL) {
-        return FALSE;
+    if (pcmk_is_set(flags, pcmk__str_star_matches) &&
+        g_list_find_custom(lst, (gpointer) "*", (GCompareFunc) strcmp)) {
+        return TRUE;
     }
```

Comment:
I think we should implement pcmk__str_star_matches in pcmk__strcmp() as well, for consistency.

Actually I'm thinking at this point, it will be better to iterate through the list ourselves and use pcmk__str_eq(), rather than g_list_find_custom(). Then we are implementing the flags in just one place, and it might be more efficient, too.

Reply:
Yeah, this seems like the best way to me too.  Fixed.